### PR TITLE
add api doc for the current API

### PR DESCRIPTION
**Pull request by kutaycandan**
Add API Doc with API Blue Print standard. Also create doc and add link in md file

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (YmirKhang) [inline .idea/misc.xml:2]: I dont think these files should be added.

File: .idea/modules.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
```
Comment (YmirKhang) [inline .idea/modules.xml:4]: this

File: .idea/vcs.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
```
Comment (YmirKhang) [inline .idea/vcs.xml:4]: also this

**Review by YmirKhang — CHANGES_REQUESTED** [review]

File: .idea/misc.xml (hunk 1 above)
Comment (YmirKhang) [reply .idea/misc.xml:2]: All the files under .idea should be deleted.

File: .idea/vcs.xml (hunk 3 above)
Comment (kutaycandan) [reply .idea/vcs.xml:4]: These are added wrongly, how can I delete these in a safe and easy way.

**Pull request closed without merging**